### ISSUE TITLE
Skipper-azure buffers uploaded file to memory instead of streaming it

### ISSUE THREAD

**Issue by super-iterator** [lukasreichart/skipper-azure#6]
From what I read about skipper and skipper-azure, I thought it streams the uploaded file to it's destination (Azure), yet, I monitored sails process and found it buffering the whole file in memory. I'm using the standard code:

```
req.file('pictures').upload({
      adapter: require('skipper-azure'),
      key: storageAccount,
      secret: accessKey,
      container: 'default'
    }, function(err, files) {
        if(!err) {
            return res.json(cdn + files[0].fd);
        } else {
            return res.send(err);
        }
    });
```
- Do I understand things mistakenly or there is something wrong?
- Also, one more thing I notice, after the file upload is done, the memory is not freed-up, as if the uploaded file is still buffered!!

This is very serious, I need your support on this.

**Comment by rijnhard** [comment]
this is pretty disheartening

**Comment by super-iterator** [comment]
@rijnhard exactly. To me, it also adds doubts to Node.js engine, when we put to it serious applications.

**Comment by rijnhard** [comment]
@super-iterator sorry for the VERY late reply, but don't make the mistake of confusing issues in a library for issues within a bigger ecosystem or runtime.

Each language runtime is suited to different use cases with overlaps. Node, for instance, is lightweight, rapidly evolving and strongly asynchronous. It, however, does have limitations, a number of which come from the V8 engine. Tradeoffs are a reality in software as I'm sure you are aware.

In this case, the biggest issue is NodeJS's memory limits but thats also why streams and iterators exist, both of which I'm using to process millions of files in a single Node process, the catch is they are harder to implement and in this case, aren't used or are done incorrectly, which is a library problem not a runtime problem.
